src/bramble/client/ProjectStats.js

```diff
@@ -0,0 +1,137 @@
+define([
+    // Change this to filer vs. filer.min if you need to debug Filer
+    "thirdparty/filer/dist/filer.min",
+],function(Filer){
+    "use strict";
+
+    var ProjectStats = {};
+    
+    var _Path = Filer.Path;
+    var _root;
+
+    // keep track of project state
+    // host cache in the object should any other project states have to be added
+    var _cache = {};
+
+    var _fs = new Filer.FileSystem();
+    var _shell = new _fs.Shell();
+    ProjectStats.getFileSystem = function(){
+        return _fs;
+    };
+    // NOTE: THIS WILL DESTROY DATA! For error cases only, or to wipe the disk.
+    ProjectStats.formatFileSystem = function(flags, callback){
+        _fs = new Filer.FileSystem(flags, callback);
```

Comment:
The reason we don't need it there, but do need it here, is that you are holding a long-lived instance of this shell, and then using it later.  If you swap out the filesystem that's bound to the shell, it will break later on.  You could stop holding this reference, and just creating it whenever you need it.